docs/apis/common/index.md

```diff
@@ -1350,3 +1350,397 @@ You may specify program arguments before the job is executed. The plan visualiza
 the execution plan before executing the Flink job.
 
 {% top %}
+
+Metrics
+-------------------
+
+Flink exposes a metric system that allows gathering and exposing metrics to external systems.
+
+### Registering metrics
+
+You can access the metric system from any user function that extends [RichFunction]({{ site.baseurl }}/apis/common/index.html#rich-functions) by calling `getRuntimeContext().getMetricGroup()`.
+This method returns a `MetricGroup` object on which you can create and register new metrics.
+
+### Metric types
+
+Flink supports `Counters`, `Gauges` and `Histograms`.
+
+#### Counter
+
+A `Counter` is used to count something. The current value can be in- or decremented using `inc()/inc(long n)` or `dec()/dec(long n)`.
+You can create and register a `Counter` by calling `counter(String name)` on a MetricGroup.
```

Comment:
Orthogonal discussion, but the name `counter` to create a counter (and others respectively) is a style break imo compared the other classes which follow Java-bean style. Is that something we should change for the release? I'm definitely fine with keeping it the way it is, but it's something I've noticed.